phoenix/app/cad_products_phoenix/lib/cad_products_phoenix_web/services/product.ex

```diff
@@ -4,34 +4,46 @@ defmodule CadProductsPhoenixWeb.Services.Product do
   alias CadProductsPhoenix.Management
   alias CadProductsPhoenix.ProductIndex
 
-  def fetch_all() do
-   Cache.get("products")
-    case Cache.get("products") do
-      {:ok, _} = result ->
-        result
-      _ ->
-        register = Management.list_register()
-        Cache.set("products", register)
-        {:ok, register}
-      end
+  def fetch_all(conn) do
+    case search_products(conn) do
+      {:ok, products} -> {:ok, products}
+      {:error, _mensage} -> {:ok, ProductIndex.get_all_product()}
+    end
+  end
+
+  def search_products(conn) do
+    register = ProductIndex.search_products(conn.params)
+    if register == [] do
+      {:error, %{code: 422, message: "Unable to search, missing params"}}
```

Comment:
A mensagem está desconexa do contexto, ao meu ver um "Items not found" seria melhor